com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/ProcessLoggerProvider.java

```diff
@@ -23,7 +23,7 @@
     private static final String DEFAULT_LOG_DIR = "logs";
     private static final String LOG_FILE_EXTENSION = ".log";
 
-    private Map<String, ProcessLogger> loggersCache = new ConcurrentHashMap<>();
+    private ConcurrentMap<String, ProcessLogger> loggersCache = new ConcurrentHashMap<>();
```

Comment:
Why is this explicitness needed? Is there any reason to go from something more abstract to something more concrete?

Reply:
ConcurrentMap overrides some methods in Map (apart from the implemented ones in ConcurrentHashMap). That's why, since it's a private variable and doesn't need to be exported, I've made this concretization. To use the benefits of a concurrent map.

Reply:
I understood the statement that it is a private variable and does not need to be exported somewhere else. But to the other statement... the Map *declared* here, has already been *initialized* as a ConcurrentHashMap, which already implements ConcurrentMap. Using this Map will make use of all the synchronization benefits. How does changing the declaration will use any more benefits? This is just the declaration type of the variable.